### PR TITLE
docs: Add flaky tests documentation

### PR DESCRIPTION
**Pull request by ghost**
This commit adds documentation for flaky tests. Flaky tests are tests that have different states with the same git sha. The documentation explains how to configure flaky tests, use the Flaky Tests UI for filtering and sorting, and perform actions such as marking tests as resolved or linking tickets. It also provides information on creating, editing, and deleting filters, as well as known issues. The documentation includes links to test reports, the Startup plan, and the Scaleup plan. Overall, this documentation aims to help users better understand and manage flaky tests in their testing pipeline.

image

### CODE REVIEW

**Comment by ralopes** [comment]
@skipi can you take a look at the documentation and see if there's something to be added?

File: docs/advanced-ci/flaky-tests.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+Flaky test detection helps identify and manage unreliable tests in a test suite.
+By detecting flaky tests, development teams can take specific actions to address
+these issues and improve the overall reliability of their testing pipeline.
+
+!!! warning "Note: A test is considered flaky when it produces different results within the same Git commit. For example, the test may fail at one point and subsequently pass without any alterations to the codebase."
+
+
+## Configuring flaky tests
+
+There are two steps needed to enable flaky tests detection, the first is to set up [test reports][test-reports-ref] in your projects,
```
Comment (csidyel) [inline docs/advanced-ci/flaky-tests.md:17]: I'd put these steps into a numbered list as below.

File: docs/advanced-ci/flaky-tests.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+3. Click Initialize.
+
+
+![zero](img/zero.png)
+
+!!! warning "Note: It may take some time for flaky tests to be detected as we begin processing your pipeline runs."
+
+##  Flaky Tests UI
+
+In the Flaky Tests UI, you can filter, sort, mark tests as resolved, link tickets, and access a detailed view for deeper analysis..
```
Comment (csidyel) [inline docs/advanced-ci/flaky-tests.md:33]: double dots at the end of the sentence.

File: docs/advanced-ci/flaky-tests.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+![zero](img/ui.png)
+
+### Filtering
+
+The Flaky Tests UI provides multiple filtering options, allowing you to refine results based on criteria such as status, date, and test name and many others
```
Comment (csidyel) [inline docs/advanced-ci/flaky-tests.md:39]: "date, and test name and many others" -> "date, test name, and many others"

**Review by Charismatron — APPROVED** [review]
Copy edited. Pull request made.

**Pull request merged**